### PR TITLE
Merge Gear Align

### PR DESCRIPTION
**Pull request by oldgalileo**
- Added new gear aligning for 2017
- Reworked RobotMap to make more sense for autonomous
- Added vision system to support gear align

### CODE REVIEW

**Review by carturn — CHANGES_REQUESTED** [review]
Please remove .DS_Store
Also, replacing NathanGain with DefaultDriver is probably not good, especially as DefaultDriver is empty.

**Review by carturn — APPROVED** [review]

**Comment by carturn** [comment]
Please make more relevant commit messages though.

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 1]
```diff
@@ -23,6 +28,7 @@
  * floating around.
  */
 public class RobotMap {
+
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Please remove.

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 2]
```diff
@@ -133,10 +101,13 @@ public RobotMap() {
...
-		HumanInput.Operator.stick = new CustomJoystick(Port.HumanInput.joystick);
-		HumanInput.Operator.stick.setDeadzone(Constant.HumanInput.OPERATOR_JOYSTICK_MINIMUM_THRESHOLD);
-		HumanInput.Driver.xbox = new CustomXbox(Port.HumanInput.xboxController);
-		HumanInput.Driver.xbox.setDeadZone(RobotMap.Constant.HumanInput.XBOX_MINIMUM_THRESHOLD);
+		Component.operatorStick = new CustomJoystick(Port.HumanInput.joystick);
+		Component.operatorStick.setDeadzone(DefaultOperator.JOYSTICK_MIN_THRESH);
+		Component.driverXbox = new CustomXbox(Port.HumanInput.xboxController);
+		Component.driverXbox.setDeadZone(DefaultDriver.CONTROLLER_MIN_THRESH);
+		// Main Subsystems
+		Component.mainSubsystems = new Subsystem[] {};
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: I think I messed up this merge, we define mainSubsystems twice.

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/DefaultDriver.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package org.usfirst.frc4904.robot.humaninterface.drivers;
+
+
+import org.usfirst.frc4904.standard.humaninput.Driver;
+
+public class DefaultDriver extends Driver {
+
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/DefaultDriver.java]: Please remove this blank line.

File: src/org/usfirst/frc4904/robot/humaninterface/operators/DefaultOperator.java [hunk 4]
```diff
@@ -4,6 +4,9 @@
 import org.usfirst.frc4904.standard.humaninput.Operator;
 
 public class DefaultOperator extends Operator {
+
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/humaninterface/operators/DefaultOperator.java]: Please remove this blank line.

File: src/org/usfirst/frc4904/robot/vision/AligningCamera.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package org.usfirst.frc4904.robot.vision;
+
+
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+
+public class AligningCamera implements PIDSource {
+
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/vision/AligningCamera.java]: Please remove this blank line.

File: src/org/usfirst/frc4904/sovereignty/strategies/GearAlign.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+public class GearAlign extends CommandGroup implements ChassisController {
+	
+	// PID Constants
+	public static double ANGLE_P = -0.0015;
+	public static double ANGLE_I = -1.0E-5;
+	public static double ANGLE_D = 0.015;
+	public static final double ANGLE_SETPOINT = 320;
+	public static final double ANGLE_TOLERANCE = 0.001;
+	//
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/sovereignty/strategies/GearAlign.java]: Please remove.

**Review by c0d3rman — CHANGES_REQUESTED** [review]
DefaultDriver being empty is meh but we'll take care of it in a later PR. Please make the changes I've requested.

**Review by c0d3rman — CHANGES_REQUESTED** [review]
Can we please merge frame dropping separately? It's untested and I would like to merge basic gear align ASAP. Move it to a new branch please.

**Review by c0d3rman — APPROVED** [review]

File: .classpath [hunk 7]
```diff
@@ -5,5 +5,6 @@
 	<classpathentry kind="var" path="networktables" sourcepath="networktables.sources"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
 	<classpathentry kind="lib" path="lib/navXLibrary/navx_frc.jar" sourcepath="lib/navXLibrary/navx_frc-sources.jar"/>
+	<classpathentry kind="lib" path="/Users/yonlern/wpilib/user/java/lib/CTRLib.jar"/>
```
Comment (carturn) [inline .classpath]: This should be `<classpathentry kind="lib" path="USERLIBS_DIR/CTRLib.jar"/>`

**Pull request merged**